Traditional chinese

kairstenfay opened the pull request:
Add traditional Chinese (zh-Hant) report translation.
Update external links to use translations where available.

kairstenfay committed:
Duplicate English report for traditional Chinese

Duplicate the English SCAN results template in preparation for a new
language: traditional Chinese.

kairstenfay committed:
Translate traditional Chinese SCAN report

Include a new Debian package, `fonts-noto-cjk`, in the Dockerfile. Note
that the Docker image now takes much longer to generate. To improve the
speed here, the Docker file could download the single language file(s)
it needs (e.g. Noto Sans CJK TC), or we could download them into this
repo.

kairstenfay committed:
zh-Hans: Use available traditional Chinese link

There's an available traditional Chinese translation for one of the WA
DoH links at the bottom of the result report. Incorporate this link into
the simplified Chinese report.

kairstenfay committed:
Translate "Never tested" for es, vi, zh-Hans

Translate the phrase "Never tested" for the three extra languages that
are already in production.

kairstenfay committed:
Replace CDC link with appropriate language

Replace the English CDC landing page with that for Spanish, Vietnamese,
and Chinese for each of the translations.

kairstenfay committed:
es: Link to Spanish WA DoH landing page

Replace the English WA DoH landing page with the one for Spanish in the
Spanish template.

kairstenfay committed:
zh: Replace WA DoH landing page with COVID-19 info

The WA DoH site is not translated into Chinese in totality (like it is
for Spanish). However, there is a landing page for COVID-19 info that is
all in Chinese. Replace the English WA DoH landing page with the Chinese
page for COVID-19 info.

kairstenfay committed:
vi: Replace WA DoH landing page with COVID-19 info

The WA DoH site is not translated into Vietnamese in totality (like it
is for Spanish). However, there is a landing page for COVID-19 info that
is all in Vietnamese. Replace the English WA DoH landing page with the
Vietnamese page for COVID-19 info.

kairstenfay commented:
![](https://github.trello.services/images/mini-trello-icon.png) [New language: Traditional Chinese](https://trello.com/c/x10272OV/552-new-language-traditional-chinese)

kairstenfay commented:
![](https://github.trello.services/images/mini-trello-icon.png) ["never tested" for all languages](https://trello.com/c/WnW5D614/555-never-tested-for-all-languages)

tsibley reviewed (approved):
LGTM assuming our language reviewers give the thumbs up. The updates to existing templates will require us to manually trigger a whole new PDF regen on the backoffice, yeah?

joverlee521 reviewed (changes requested):
I would definitely recommend increasing the font size for traditional Chinese. Even I have squint to read it with the current font size.

kairstenfay commented:
> LGTM assuming our language reviewers give the thumbs up. The updates to existing templates will require us to manually trigger a whole new PDF regen on the backoffice, yeah?

We're pending review now. 

Yes, someone will have to manually create new PDFs for zh-Hant. This task is tracked in the Trello card.

kairstenfay commented:
> I would definitely recommend increasing the font size for traditional Chinese. Even I have squint to read it with the current font size.

Thank you for this feedback!

kairstenfay committed:
zh-Hans: Use NotoSans font instead of Fandol

Instead of using the default font (Fandol) from the xeCJK package, use
NotoSans like we've done for zh-Hant.

Because we're downloading the `fonts-noto-cjk` package, remove
`texlive-lang-chinese` from the Dockerfile.

kairstenfay commented:
`zh-Hant` now has a working bold-face font (thanks for catching this, @joverlee521 !). Unfortunately this required the download of Noto Sans CJK fonts which, on my machine, slows down the Docker image generation by about 2 minutes.

kairstenfay commented:
I just force-pushed after rewording a commit. No code changes.

kairstenfay committed:
zh: Use locally downloaded Noto fonts

Instead of downloading the giant package `fonts-noto-cjk`, download the
four specific fonts we use for Chinese translations: regular and bold
simplified and traditional Chinese. Inside the results template, map the
font to the new, local `fonts` directory accessible by Docker when the
current directory is mounted (via `-v`).

kairstenfay commented:
Just pushed a new commit with downloaded Chinese fonts in the `fonts` directory. The Docker image is no longer burdened by downloading the huge Noto CJK package.

kairstenfay commented:
I just force-pushed a reword and a minor whitespace edit. I don't intend to make further changes at this time.

tsibley commented on scan/report-zh-Hans.tex:
```diff
@@ -28,8 +28,9 @@
-\setCJKmainfont{NotoSansCJK}[
-  Extension   = .ttc,
+\setCJKmainfont{NotoSansCJKsc}[
+  Path        = /output/fonts/,
```
Hardcoding this Path means there's now a hard requirement on a `/output` mount and no chance of this working outside of Docker. I think instead the font files could be copied into the Docker image and referred to by a relative path in these templates.

tsibley replied:
Thinking about this a bit more: we're using Noto Sans CJK now for Simplified Chinese, Traditional Chinese, and in the future (presumably) Korean. That just leaves Japanese as the only language which is in the larger `fonts-noto-cjk` package but that we won't use, so maybe the returns are dwindling.

I think the only remaining win might be if individual PDF generation (not Docker image generation) is faster when TeX only has to load the single-language font files. Not sure about this.

kairstenfay commented:
Rebased onto master.
The zh-Hant report has been verified by a native speaker.

tsibley reviewed (commented):
Somehow I never submitted the comments below. Just found them because discussion in Slack triggered my memory. Submitting now for posterity, even though this PR is merged already.
